sound/soc/sof/intel/hda-stream.c

```diff
@@ -312,6 +297,24 @@ int hda_dsp_stream_trigger(struct snd_sof_dev *sdev,
 		hstream->running = false;
 		snd_sof_dsp_update_bits(sdev, HDA_DSP_HDA_BAR, SOF_HDA_INTCTL,
 					1 << hstream->index, 0x0);
+
+#if !IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_ALWAYS_ENABLE_DMI_L1)
+		/* check if there are any capture streams running */
+		list_for_each_entry(s, &bus->stream_list, list) {
+			if (s->running &&
+			    s->direction == SNDRV_PCM_STREAM_CAPTURE) {
+				active_capture_stream = true;
+				break;
+			}
+		}
+
+		/* Enable DMI L1 entry if there's no running capture stream */
+		if (!active_capture_stream)
+			snd_sof_dsp_update_bits(sdev, HDA_DSP_HDA_BAR,
+						HDA_VS_INTEL_EM2,
+						HDA_VS_INTEL_EM2_L1SEN,
+						HDA_VS_INTEL_EM2_L1SEN);
+#endif
```

Comment:
so why do you need to go through the list if if all the streams are disabled on TRIGGER_STOP? This code was fine with the previous solution, it makes no sense here, all you need is test if this is a capture stream

Reply:
the trigger_stop is for specific stream only, not for all streams. The point here is:
If we are stopping one capture stream, let's check if it is the last running capture stream, if it is, then let's restore L1 enable bit.